packages/android-client/FishjamClient/src/main/java/com/fishjamcloud/client/CommandsQueue.kt

```diff
@@ -41,9 +44,20 @@ internal class CommandsQueue {
     }
   }
 
-  fun clear(cause: String) {
+  fun onDisconnected() {
     clientState = ClientState.CREATED
-    commandsQueue.forEach { command -> command.job.cancel(CancellationException(cause)) }
+    commandsQueue
+      .filter { it.isConnectionCommand() }
+      .forEach { command -> command.continuation?.resume(Unit) }
+    commandsQueue.removeAll { it.isConnectionCommand() }
+    if (commandsQueue.isNotEmpty()) {
+      commandsQueue.first().execute()
+    }
```

Comment:
Can you explain why all connection commands are removed and then we execure remaining commend? Also what if there is more than one command to execute?

Reply:
we can add/remove tracks when there is no connection, it would be weird to silently not add track because the connection was broken and we don't like throwing errors now
we can't however for example join 

if there is more than one command it will be executed automatically by the queue when the first command finishes